strawberryfields/compilers/gaussian_unitary.py

```diff
@@ -18,18 +18,62 @@
 from strawberryfields import ops
 from strawberryfields.parameters import par_evaluate
 from thewalrus.symplectic import (
-    expand_vector,
     expand,
     rotation,
     squeezing,
     two_mode_squeezing,
     interferometer,
     beam_splitter,
 )
-
+from numba import jit
 from .compiler import Compiler
 
 
+@jit(nopython=True)
+def _apply_symp_one_mode_gate(S_G, S, r, i):
+    """In-place applies a one mode gate G to the symplectic operation, S, in mode i
+    Args:
+        S_G (array): 2x2 matrix for one mode symplectic operation
+        S (array): Symplectic operation
+        i (int): index of one mode gate
+    """
+    M = S.shape[0] // 2
+    (S[i], S[i + M]) = (
+        S_G[0, 0] * S[i] + S_G[0, 1] * S[i + M],
+        S_G[1, 0] * S[i] + S_G[1, 1] * S[i + M],
+    )
+    (r[i], r[i + M]) = (
+        S_G[0, 0] * r[i] + S_G[0, 1] * r[i + M],
+        S_G[1, 0] * r[i] + S_G[1, 1] * r[i + M],
+    )
+    return S, r
+
+
+@jit(nopython=True)
+def _apply_symp_two_mode_gate(S_G, S, r, i, j):
```

Comment:
```suggestion
def _apply_symp_two_mode_gate(S_G, S, r, i, j): # pragma: no cover
```

Reply:
These two `#pragma: no cover` are due to the fact that `pytest` does not play nicely with jitted code when preparing coverage reports, so we have to exclude them from the coverage report.

Reply:
The other option is to not jit these functions. Could we use useful to quickly benchmark how much difference it makes adding the jit.